src/Raven.Client/Documents/Operations/Indexes/DisableIndexOperation.cs

```diff
@@ -2,42 +2,82 @@
 using System.Net.Http;
 using Raven.Client.Documents.Conventions;
 using Raven.Client.Http;
+using Raven.Client.Json;
+using Raven.Client.Util;
 using Sparrow.Json;
 
 namespace Raven.Client.Documents.Operations.Indexes
 {
     public class DisableIndexOperation : IMaintenanceOperation
     {
-        private readonly string _indexName;
+        private readonly Parameters _parameters;
 
         public DisableIndexOperation(string indexName)
         {
-            _indexName = indexName ?? throw new ArgumentNullException(nameof(indexName));
+            if (indexName == null)
+                throw new ArgumentNullException(nameof(indexName));
+
+            _parameters = new Parameters
+            {
+                IndexName = indexName,
+                ClusterWide = false
+            };
+        }
+
+        public DisableIndexOperation(string indexName, bool clusterWide)
+        {
+            if (indexName == null)
+                throw new ArgumentNullException(nameof(indexName));
+
+            _parameters = new Parameters
+            {
+                IndexName = indexName,
+                ClusterWide = clusterWide
+            };
         }
 
         public RavenCommand GetCommand(DocumentConventions conventions, JsonOperationContext context)
         {
-            return new DisableIndexCommand(_indexName);
+            return new DisableIndexCommand(conventions, context, _parameters);
         }
 
-        private class DisableIndexCommand : RavenCommand
+        private class DisableIndexCommand : RavenCommand, IRaftCommand
         {
-            private readonly string _indexName;
+            private readonly BlittableJsonReaderObject _parameters;
 
-            public DisableIndexCommand(string indexName)
+            public DisableIndexCommand(DocumentConventions conventions, JsonOperationContext context, Parameters parameters)
             {
-                _indexName = indexName ?? throw new ArgumentNullException(nameof(indexName));
+                if (conventions == null)
+                    throw new ArgumentNullException(nameof(conventions));
+                if (context == null)
+                    throw new ArgumentNullException(nameof(context));
+                if (parameters == null)
+                    throw new ArgumentNullException(nameof(parameters));
+
+                _parameters = DocumentConventions.Default.Serialization.DefaultConverter.ToBlittable(parameters, context);
             }
 
             public override HttpRequestMessage CreateRequest(JsonOperationContext ctx, ServerNode node, out string url)
             {
-                url = $"{node.Url}/databases/{node.Database}/admin/indexes/disable?name={Uri.EscapeDataString(_indexName)}";
+                url = $"{node.Url}/databases/{node.Database}/admin/indexes/disable";
```

Comment:
This will break backward comp. for studio and users .(I don't think we care about the studio since, we need to re-work it)
Although I like the parameters approach here we need to maintain the back comp.

cc @ppekrol

Reply:
You need to send the name anyway. I think it would be better just to include both of them in the query string.